### ISSUE TITLE
[Feature request] : handle links

### ISSUE THREAD

**Issue by ghost** [CanisLupus/swift-selection-search#120]
_Swift Selection Search_  3.27.2 / Firefox 62.0.3 (x64) running smoothly.

I'm facing a handicap when it comes to links in which case I have to hover them with the ALT key ("delinkify") to have SSS popup. At this time I use a dedicated extension called _Redirect Link_ with a lot of engines in a similar way to SSL.

For instance, a shortened url link that I wish to check for the true destination by sending it to a site such as _CheckShortURL_ : if SSS would handle links I could check it immediately (given of course the search engine be included in my SSS settings).

Thanks!

**Comment by CanisLupus** [comment]
Hey! :) So, if I'm understanding this correctly, you'd like SSS to recognize that the text you selected is inside a link and provide search engines or other functions specifically for links. Is that right?

I think this would need code support similar to what is needed for issue #93 (where the copy icon would be one of the options for the link).

What would make sense? To have an option for each individual engine where you could specify that it should only appear for links? (otherwise it appears for general text as usual). I'd like to hear more about your specific use-case if you have other things in mind. :)

Either way, this is so that I can think about what the implementation needs, for whenever I'm able to work on SSS again. Issue #106 is up next. ;)

Cheers!
Daniel

**Comment by StanGets** [comment]
Hi @CanisLupus ! I conceptualize correctly but I lack to explain accordingly :)
The truth is I had switched to SSS after having been a user of a similar extension, _ContextSearch web-ext_ ... and I faced then the similar drama (!) and related it at the time at https://github.com/ssborbis/ContextSearch-web-ext/issues/11 ... maybe had I then better explained the suggestion (I appear as ghost because I had closed my account since then).

As for the way of having urls handled by SSS I'd conceive what had been done by _ContextSearch web-ext_ which is to keep the frame as it is (no special category) but to simply consider urls, which would mean of course that the extension be triggered by urls as well as by text (but with a right-click on links which would either open the traditional menu with SSS included or SSS' popup, it's up to you to see). From there on the user would of course have to apply correctly his search engines.

I think it remains an interesting enhancement for SSS, but I'm not the developer. But it's feasible I guess.

**Comment by CanisLupus** [comment]
Aaah, I see now. So you'd like SSS to detect right clicks on links and show its context menu for them (while currently it only does so if text is selected).

I didn't understand at first because I tend to think of SSS as _primarily_ the popup and only then the context menu, although some people do prefer the latter. :)

I'll consider this for a future update, it makes a lot of sense.

**Comment by StanGets** [comment]
I prefer the popup personally, I hardly never use SSS via the context-menu. But should the link detection be included in SSS then I'm not sure that right-clicking bringing up the SSS popup would be wise, especially that right-clicking (text, link) is associated to the context-menu... anyway my suggestion, as far as i'm concerned, if it would indeed ease search on links concerns nevertheless a very small minority of my quests which practically always concern text. The suggestion is meant to bring a "plus" and is far from corresponding to a need.

**Comment by CanisLupus** [comment]
Yeah, I generally use the popup myself. :) Right clicking to open the popup would indeed create incompatibilities with general browser usage.

By making use of the Alt key to "delinkify" (by the way, I didn't know that was a thing that existed, so thank you ;)), the user could easily select some text on the link and I could implement what I mentioned in my first reply. It would be more complicated, but doable. I'll have to think about these other approaches (probably not double click since it's not very "standard"), but at least the context menu solution for this problem should be simple enough.
